core/init.js

```diff
@@ -1,21 +1,45 @@
+/*
+    Bootstraps the framework. Projects should consume this module and wait for
+    the exposed promise to resolve before doing anything else.
+
+    helpers -- sets up Nunjucks globals, helpers, filters.
+    l10n_init -- loads locale script, sets up w.nav.l10n.
+    navigation -- sets up navigation event handlers.
+    polyfill -- sets up browser polyfills.
+
+    Also configures some things to bootstrap the framework.
+
+    - Adds base routes such as login.
+    - Makes ?lang parameter is sent out with each XHR.
+*/
 define('core/init',
-    ['core/helpers', 'core/navigation', 'core/polyfill', 'core/router',
-     'core/utils'],
-    function(helpers, navigation, polyfill, router,
-             utils) {
-
-    router.addRoutes([
-        {'pattern': '^/fxa-authorize$', 'view_name': 'core/fxa_authorize'},
-    ]);
-
-    router.api.addRoutes({
-        'fxa-login': '/api/v2/account/fxa-login/',
-        'login': '/api/v2/account/login/',
-        'logout': '/api/v2/account/logout/',
-        'site-config': '/api/v2/services/config/site/?serializer=commonplace',
-    });
-
-    router.api.addProcessor(function(endpoint) {
-        return {lang: utils.lang()};
-    });
+    ['core/helpers', 'core/l10n_init', 'core/navigation', 'core/polyfill',
+     'core/router', 'core/utils'],
+    function(helpers, l10n_init, navigation, polyfill,
+             router, utils) {
+
+    function init() {
```

Comment:
I think we should add [squire](https://github.com/iammerrick/Squire.js/) rather than wrapping this in a function. I tried not to add squire before because it was slower but I think it's better than modifying our code everywhere for testing.

With squire you could require new versions of the modules you want to test which is a bit verbose but probably more reliable.

```js
it('sets some routes', function(done) {
    new Squire().require(['require', 'core/router'], function(require, router) {
        assert.equal(router.routes.length, 0);
        assert.equal(router.api.routes.length, 0);
        require(['core/init'], fuction() {
            assert(router.routes.length > 0);
            assert(router.api.routes.length > 0);
            done();
        });
    });
}));
```